src/services/SendEmailService.ts

```diff
@@ -72,16 +72,34 @@ export class SendEmailService {
 	async sendEmail(message: Email): Promise<EmailResponse> {
     	let encoded;
     	// Fetch the instructions pdf from Yoti
-    	this.logger.debug("Fetching the Instructions Pdf from yoti for sessionId: ", message.yotiSessionId);
-    	try {
-    		const instructionsPdf = await this.yotiService.fetchInstructionsPdf(message.yotiSessionId);
-			if (instructionsPdf) {
-				encoded = Buffer.from(instructionsPdf, "binary").toString("base64");
+		let yotiInstructionsPdfRetryCount = 0;
+		//retry for maxRetry count configured value if fails
+		while (yotiInstructionsPdfRetryCount++ < this.environmentVariables.yotiInstructionsPdfMaxRetries() + 1) {
+			this.logger.debug("Fetching the Instructions Pdf from yoti for sessionId: ", message.yotiSessionId);
+			try {
+				const instructionsPdf = await this.yotiService.fetchInstructionsPdf(message.yotiSessionId);
+				if (instructionsPdf) {
+					encoded = Buffer.from(instructionsPdf, "binary").toString("base64");
+					break;
+				}
+			} catch (err: any) {
+				this.logger.error("Error while fetching Instructions pfd or encoding the pdf.", { err });
+				if (err.statusCode === 500 || err.statusCode === 429) {
+					if (yotiInstructionsPdfRetryCount < this.environmentVariables.yotiInstructionsPdfMaxRetries() + 1) {
```

Comment:
is this check needed or we could log the message and sleep for backoff period only 🤔